scripts/test.sh

```diff
@@ -20,28 +20,42 @@ source "${PROG_DIR}/.util/tools.sh"
 source "${PROG_DIR}/.util/print.sh"
 
 function main() {
-  local clean
+  local clean regport dont_create_registry
   clean="false"
+  regport=""
+  dont_create_registry="false"
```

Comment:
I wonder if we made this create_registry with a default of `false` and then set it to true as an optional argument or something like that, it might make integration with the shared version of test.sh easier later on?

Reply:
You mean instead of naming it `dont_create_registry` naming it `create_registry` and set it to `false`. I agree with changing the name, although the default behaviour I would like to be create the registry (setting `create_registry=true`), as it also matches how jammy calls it on the integration tests (without passing any arguments) https://github.com/paketo-buildpacks/jammy-base-stack/blob/f96f2f10ee3a438657442763bae24ee8ba9bbaf7/.github/workflows/test-pull-request.yml#L26

Reply:
>  as it also matches how jammy calls it on the integration tests (without passing any arguments)

I think it comes down to in how many places where the shared scripts/test.sh gets called a registry is needed. Ideally we'd want the default to match the majority. If passing arguments is the concern maybe there is another way to signal that a registry is needed. Maybe by adding another file in the repo and having start check for that to decide if a registry is needed or not ?

Reply:
I've aligned tools.sh and test.sh as much as possible matching the majority of the implementations. The current implementation only affects the test.sh script and not how the test.sh script is being called, as a result it matches with the current github workflows. There is no need to pass any argument ATM to create the registry, as if there is not one it creates it otherwise it uses the existing one.